src/vs/workbench/browser/parts/editor/textResourceEditor.ts

```diff
@@ -150,6 +150,34 @@ export class AbstractTextResourceEditor extends BaseTextEditor {
 		}
 	}
 
+	/**
```

Comment:
Having this helper method in `textResourceEditor` does not really make sense to me as it would only be used by the output.
I would do all this logic in the outputService as all this API should be public.